tests/components/volvo/test_config_flow.py

```diff
@@ -0,0 +1,257 @@
+        config_flow["flow_id"], {CONF_API_KEY: "abcdef0123456879abcdef"}
+    )
+
+    assert len(hass.config_entries.async_entries(DOMAIN)) == 0
+    assert config_flow["type"] is FlowResultType.FORM
+    assert config_flow["errors"]["base"] == "cannot_load_vehicles"
+    assert config_flow["step_id"] == "api_key"
+
+    config_flow = await _async_run_flow_to_completion(
+        hass, config_flow, aioclient_mock, configure=False
+    )
+    assert config_flow["type"] is FlowResultType.CREATE_ENTRY
+
+
+async def _async_run_flow_to_completion(
+    hass: HomeAssistant,
+    config_flow: ConfigFlowResult,
+    aioclient_mock: AiohttpClientMocker,
+    *,
+    configure: bool = True,
+    has_vin_step: bool = True,
+    is_reauth: bool = False,
+    api_key_failure: bool = False,
+) -> ConfigFlowResult:
+    if configure:
+        await _mock_api_client(hass, aioclient_mock, api_key_failure=api_key_failure)
+        config_flow = await hass.config_entries.flow.async_configure(
+            config_flow["flow_id"]
+        )
+
+    if is_reauth and not api_key_failure:
+        return config_flow
+
+    assert config_flow["type"] is FlowResultType.FORM
+    assert config_flow["step_id"] == "api_key"
+
+    await _mock_api_client(hass, aioclient_mock, api_key_failure=False)
+    config_flow = await hass.config_entries.flow.async_configure(
+        config_flow["flow_id"], {CONF_API_KEY: "abcdef0123456879abcdef"}
+    )
+
+    if has_vin_step:
+        assert config_flow["type"] is FlowResultType.FORM
+        assert config_flow["step_id"] == "vin"
+
+        config_flow = await hass.config_entries.flow.async_configure(
+            config_flow["flow_id"], {CONF_VIN: DEFAULT_VIN}
+        )
+
+    return config_flow
+
+
+async def _mock_api_client(
+    hass: HomeAssistant,
+    aioclient_mock: AiohttpClientMocker,
+    *,
+    single_vin: bool = False,
+    api_key_failure: bool = False,
+) -> None:
+    aioclient_mock.clear_requests()
+    aioclient_mock.post(
+        TOKEN_URL,
+        json=SERVER_TOKEN_RESPONSE,
+    )
+
+    vins = [{"vin": DEFAULT_VIN}]
+
+    if not single_vin:
+        vins.append({"vin": "YV10000000AAAAAAA"})
+
+    if api_key_failure:
+        aioclient_mock.get(
+            f"{_API_URL}{_API_CONNECTED_ENDPOINT}",
+            exc=VolvoApiException(),
+        )
+    else:
+        aioclient_mock.get(
+            f"{_API_URL}{_API_CONNECTED_ENDPOINT}",
+            json={
+                "data": vins,
+            },
+        )
+
+    vehicle_data = await async_load_fixture_as_json(hass, "vehicle", DEFAULT_MODEL)
+    aioclient_mock.get(
+        f"{_API_URL}{_API_CONNECTED_ENDPOINT}/{DEFAULT_VIN}",
+        json={
+            "data": vehicle_data,
+        },
+    )
+
+    vehicle_data = await async_load_fixture_as_json(hass, "vehicle", "xc90_petrol_2019")
+    aioclient_mock.get(
+        f"{_API_URL}{_API_CONNECTED_ENDPOINT}/YV10000000AAAAAAA",
+        json={
+            "data": vehicle_data,
+        },
+    )
```

Comment:
Ideally we would just mock out the object like we do in the other tests to stay consistent